lighthouse-viewer/app/src/lighthouse-report-viewer.js

```diff
@@ -164,12 +169,30 @@ class LighthouseReportViewer {
       } catch (e) {
         throw new Error('Could not parse JSON file.');
       }
-
       this._reportIsFromGist = false;
       this._replaceReportHtml(json);
     }).catch(err => logger.error(err.message));
   }
 
+  /**
+   * Opens new tab with compatible report viewer version
+   * @param {!ReportRenderer.ReportJSON} reportJson
+   * @private
+   */
+  _loadInLegacyViewerVersion(json) {
+    const warnMsg = `Version mismatch between viewer and JSON.
+    Opening new tab with compatible viewer.`;
+    const features = new ViewerUIFeatures(new DOM(document));
+    const viewerPath = '/lighthouse/viewer2x/';
+
+    logger.log(warnMsg, false);
+    features.sendJsonReport(json, viewerPath, onMessage);
```

Comment:
impl is a bit confusing that we create a temporary instance of the ViewerUIfeatures to trigger the render in viewer, feels a bit hacky

I think it's mostly just the fake flexibility of `sendJsonReport` that confuses me. WDYT of making this a bit more explicit with a `sendJsonReportToV2` or something that both sure the same underlying logic?

they could just resolve a promise when they get confirmation that the report has been rendered and nix the onMessage piece